### DISCUSSION TITLE
Used clients' ports

### DISCUSSION THREAD

**Discussion by mostafa0017** [rustdesk/rustdesk#8043] [Q&A]
I am cross posting here from reddit.

I tried to use RustDesk using the public relays, one client behind Sophos Firewall the other is not, couldn't get to the password field at 
all and had to whitelist TCP `21114-21119` and UDP `21116`.
After that the password field appeared, I entered the password and bam I get a "reset by the peer" error, after inspecting Sophos packet 
capture directed at the client behind Sophos I found this:

| Source IP | Destination IP | Packet type | Ports [src,dst] | Status |
| ---------  | -------------  | -------------  | -------------  | ------------- |
| 10.0.5.73 | External Client IP | TCP | 61434,35578 | Violation |
| 10.0.5.73 | External Client IP | TCP | 61434,35578 | Incoming |
| 10.0.5.73 | 108.61.171.103 | TCP | 61434,21116 | Incoming |

So RustDesk does use the range `21114-21119` while the client is communicating with the relay but after that when the clients talk to each other they use random ports, for example the following ports were tried sequentially

| Ports [src,dst] |
| ------------- |
| 61440,35591 |
| 61437,35581 |
| 61434,35578 |

Any way to control that ? As I can't whitelist all ports just like that.

**Comment by mostafa0017** [comment]
@rustdesk Any comment on this ?

**Comment by jimisens** [comment]
I'm currently using RustDesk to offer a maintenance service to a client with multiple stores. My client has a firewall that controls the computers' Internet access.
We can access certain stores remotely without any problem but others there is no way.
My client asked me for the ports he should open and I gave him the ones announced by rustdesk
TCP (21114, 21115, 21116, 21117, 21118, 21119)
UDP (21116)

However, even opening these ports in the firewall, we still cannot connect to the computers.

I think the problem lies precisely in the fact that the rustdesk client uses random ports and these are blocked by the firewall.
Is there a way to define this port range to allow rustdesk to work in places controlled by firewalls?

**Comment by xinglicai** [comment]
1. Open the rustdesk.toml file in the rustdesk configuration directory

C:\Users\Current Username\AppData\Roaming\RustDesk\config

2. Use a text editor to add at the end

[relay]

port_range = "21114-21119"

3. Exit rustdesk in the status bar in the lower right corner

4. Reopen rustdesk and click Start Service

**Reply by jimisens** [reply]
Thank you @xinglicai 
I will try this
